README.md

```diff
@@ -1,8 +1,129 @@
+
+**Why caramel?** Compared to Sux4J, we offer about 25% faster construction, a much smaller and simpler codebase (> 10k vs 2k LoC for the core implementation), and easy Python bindings. We support arbitrary data types for keys and values, enabling more flexibility for applications (the original implementation supports only string keys and integer values). We also have some algorithmic improvements to make CSF-backed tables practical for real applications. For example, we have optimal-size Bloom prefilters for situations when one value is very common. We also have value-permutation methods to preprocess 2D arrays into a more compressible format, as well as tricks to implement ragged arrays and other lookup primitives using CSFs. Our paper describes these tricks and applications.
+
+### Minimal API Example
+
+You can use our Python API like this.
+
+```python
+import caramel
+keys = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J"]
+values = [0, 0, 0, 1, 2, 0, 1, 4, 0, 1]
+# Constructing
+csf = caramel.CSF(keys, values)
+# Querying
+csf.query("D")  # Returns 1
+# Saving
+csf.save("map.csf")
+# Loading
+csf = caramel.load("map.csf")
+```
+
+## Getting Started
+The recommended way to interact with our library is through the Python bindings, though we also offer a C++ API. 
+
+### Python Installation
+To install the Python bindings, first clone the repository and all of our dependencies.
+
+```bash
+git clone --recurse-submodules git@github.com:detorresramos/caramel.cpp.git
+cd caramel.cpp
+```
+
+Next, run our build script:
+```bash
+python3 bin/build.py
+```
+
+You should see a significant amount of CMake build output, (hopefully) ending with a successful installation!
+```
+... build output ...
+  -- Configuring done
+  -- Generating done
+  -- Build files have been written to: /Users/bencoleman/Documents/code/src/caramel.cpp/build
+  [  4%] Built target libomp-needed-headers
+  Consolidate compiler generated dependencies of target omp
+  [ 74%] Built target omp
+  Consolidate compiler generated dependencies of target caramel_lib
+  [ 95%] Built target caramel_lib
+  Consolidate compiler generated dependencies of target _caramel
+  [ 97%] Building CXX object CMakeFiles/_caramel.dir/python_bindings/PythonBindings.cc.o
+  [100%] Linking CXX shared module lib.macosx-12-arm64-cpython-311/caramel/_caramel.cpython-311-darwin.so
+  [100%] Built target _caramel
+... more build output ...
+Successfully installed caramel-0.0.1
+```
+
+**Tip:** If you use Anaconda or virtual environments to manage your Python environments, be sure to switch to the correct environment before calling `bin/build.py`. Our build script will build and install the Python module into *your current Python environment!*
+
+**Build dependencies:** We require Python >= 3.8 and a C++ compiler supporting C++20. Older versions may build, but we can't guarantee support.
+
+### Testing your installation
+To run the Python tests, you can run
```

Comment:
I can add some openmp install instructions. Do you happen to remember what you did when first installing